.github/workflows/test.yml

```
@@ -86,18 +79,100 @@ jobs:
+      only_rust: ${{ steps.check-rust-only.outputs.only_rust }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          fetch-depth: 0
+
+      # Check if changes are ONLY rust
+      - name: Check if changes are ONLY rust
+        id: check-rust-only
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:5:30: Double quote to prevent globbing and word splitting

(shellcheck)